scripts/peer-dependency-generator/generatePeerDependencyDoc.js

```diff
@@ -0,0 +1,24 @@
+const generatePeerDependencyDoc = (peerDependencies) => {
+  const tableHeader = '| Peer Dependency | Version |';
+  const tableHeaderBottom = '|-|-|';
+
+  const dependencyRows = Object.keys(peerDependencies).sort().map((dependency) => {
+    // Use the peer dependency version specified in the package.json
+    const specifiedVersion = peerDependencies[dependency];
+
+    return `| ${dependency} | ${specifiedVersion} |`;
+  });
+
+  const dependencyTableMarkdown = `\n${tableHeader}\n${tableHeaderBottom}\n${dependencyRows.join('\n')}\n`;
+
+  const peerDependenciesContent = `<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
+## Peer Dependencies
+
+This component requires the following peer dependencies be installed in your app to properly function as designed.
```

Comment:
talked about putting this within a details tag

Reply:
Updated in this commit: https://github.com/cerner/terra-core/pull/2518/commits/df81d96431f3ffd3169cc98ab5634730a895b412